agent/api/task/task_test.go

```diff
@@ -1713,3 +1715,138 @@ func TestPostUnmarshalTaskASMDockerAuth(t *testing.T) {
 	err := task.PostUnmarshalTask(cfg, credentialsManager, resFields, nil, nil)
 	assert.NoError(t, err)
 }
+
+func TestPostUnmarshalTaskSecret(t *testing.T) {
+	secret := apicontainer.Secret{
+		Provider: "ssm",
+		Name: "secret",
+		Region: "us-west-2",
+		ValueFrom: "/test/secretName",
+	}
+
+	container := &apicontainer.Container{
+		Name:  "myName",
+		Image: "image:tag",
+		Secrets: []apicontainer.Secret{secret},
+		TransitionDependenciesMap: make(map[apicontainerstatus.ContainerStatus]apicontainer.TransitionDependencySet),
+	}
+
+	task := &Task{
+		Arn:                "test",
+		ResourcesMapUnsafe: make(map[string][]taskresource.TaskResource),
+		Containers:         []*apicontainer.Container{container},
+	}
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	cfg := &config.Config{}
+	credentialsManager := mock_credentials.NewMockManager(ctrl)
+	ssmClientCreator := mock_ssm_factory.NewMockSSMClientCreator(ctrl)
+
+	resFields := &taskresource.ResourceFields{
+		ResourceFieldsCommon: &taskresource.ResourceFieldsCommon{
+			SSMClientCreator:   ssmClientCreator,
+			CredentialsManager: credentialsManager,
+		},
+	}
+
+	err := task.PostUnmarshalTask(cfg, credentialsManager, resFields, nil, nil)
+	assert.NoError(t, err)
+}
+
+func TestGetAllSSMSecretRequirements(t *testing.T) {
+	regionWest := "us-west-2"
+	regionEast := "us-east-1"
+
+	secret1 := apicontainer.Secret{
+		Provider: "ssm",
+		Name: "secret1",
+		Region: regionWest,
+		ValueFrom: "/test/secretName1",
+	}
+
+	secret2 := apicontainer.Secret{
+		Provider: "asm",
+		Name: "secret2",
+		Region: regionWest,
+		ValueFrom: "/test/secretName2",
+	}
+
+	secret3 := apicontainer.Secret{
+		Provider: "ssm",
+		Name: "secret3",
+		Region: regionEast,
+		ValueFrom: "/test/secretName3",
+	}
+
+	container := &apicontainer.Container{
+		Name:  "myName",
+		Image: "image:tag",
+		Secrets: []apicontainer.Secret{secret1, secret2, secret3},
+		TransitionDependenciesMap: make(map[apicontainerstatus.ContainerStatus]apicontainer.TransitionDependencySet),
+	}
+
+	task := &Task{
+		Arn:                "test",
+		ResourcesMapUnsafe: make(map[string][]taskresource.TaskResource),
+		Containers:         []*apicontainer.Container{container},
+	}
+
+	reqs := task.getAllSSMSecretRequirements()
+	t.Log(reqs)
+	assert.Equal(t, secret1, reqs[regionWest][0])
+	assert.Equal(t, secret3, reqs[regionEast][0])
```

Comment:
Please add an assert for secret2 as well.

Reply:
Added

Reply:
I added another assertion for the length of secrets in regionWest to show the secret2 hasn't be added.